### ISSUE TITLE
Signature verification broken

### ISSUE THREAD

**Issue by michaelkuhn** [rpm-software-management/rpm#330]
On a freshly installed Fedora 27 system (RPM 4.14.0-rc2), DNF does not prompt me to install keys even though they are missing (`f5282ee4` is the Fedora 27 key and manually importing it makes the warning go away):

```
warning: /var/cache/dnf/.../foo.rpm: Header V3 RSA/SHA256 Signature, key ID f5282ee4: NOKEY
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
...
Complete!
```

I tracked this down to `hdrFromFdno` (in `transaction.py`) not throwing the appropriate exceptions. Instead, `TransactionSetCore.hdrFromFdno` actually returns `0`, even though the above warning is produced. This leads DNF to believe that everything is fine and install the package.

**Comment by pmatilai** [comment]
Urks... yeah, reproduced, I'll look into it. Thank you for reporting!

**Comment by pmatilai** [comment]
Okay, fixed now in commit e6c88eaeab3661f5fb180b970cafba233bcd999a and testcase added to prevent future regressions.

Again, kudos & thanks for spotting and reporting! Would've been embarrassing to ship 4.14.0 with this bug...

**Comment by michaelkuhn** [comment]
I am glad I could help. :-) Thanks for the fast fix and (planned) Fedora update!
